src/main/groovy/io/nextflow/gradle/NextflowPlugin.groovy

```diff
@@ -86,11 +93,19 @@ class NextflowPlugin implements Plugin<Project> {
                 deps.testRuntimeOnly "net.bytebuddy:byte-buddy:1.14.17"
                 deps.testImplementation(testFixtures("io.nextflow:nextflow:${nextflowVersion}"))
                 deps.testImplementation(testFixtures("io.nextflow:nf-commons:${nextflowVersion}"))
+
+                // dependencies for buildIndex task
+                deps.indexFile "io.nextflow:nextflow:${nextflowVersion}"
```

Comment:
Why using a custom configuration for this? what nextflow classes are required by the plugin for the the index generation ?

Reply:
It uses `nextflow.plugin.index.PluginIndexWriter`, which is defined in the corresponding Nextflow PR https://github.com/nextflow-io/nextflow/pull/6361

Reply:
Fair enough